clang/lib/Sema/BoundsWideningAnalysis.cpp

```diff
@@ -700,18 +783,125 @@ void BoundsWideningAnalysis::GetVarsAndBoundsForModifiedVars(
 
   // Get the set of variables that are pointers to null-terminated arrays and
   // in whose lower and upper bounds expressions the modified variables occur.
-  VarSetTy NullTermPtrsWithVarsInBounds;
-  BWUtil.GetNullTermPtrsWithVarsInLowerBounds(ModifiedVars,
-                                              NullTermPtrsWithVarsInBounds);
-  BWUtil.GetNullTermPtrsWithVarsInUpperBounds(ModifiedVars,
-                                              NullTermPtrsWithVarsInBounds);
+  VarSetTy PtrsWithAffectedBounds;
+  BWUtil.GetPtrsWithVarsInLowerBounds(ModifiedVars,
+                                      PtrsWithAffectedBounds);
+  BWUtil.GetPtrsWithVarsInUpperBounds(ModifiedVars,
+                                      PtrsWithAffectedBounds);
 
   // For each null-terminated array we need to reset the bounds to its declared
   // bounds.
-  for (const VarDecl *V : NullTermPtrsWithVarsInBounds) {
+  for (const VarDecl *V : PtrsWithAffectedBounds) {
     BoundsExpr *NormalizedBounds = SemaRef.NormalizeBounds(V);
     VarsAndBounds[V] = dyn_cast_or_null<RangeBoundsExpr>(NormalizedBounds);
   }
+
+  // If the modification of a variable by the current statement affects the
+  // bounds of a null-terminated array, then check invertibility of the
+  // statement. If the statement is invertible then store the statement, the
+  // modified LValue, the original LValue and the set of null-terminated arrays
+  // whose bounds are affected by the statement. We will use this info in the
+  // computation of the Out sets of blocks.
```

Comment:
Nit: ...computation of the Out sets of blocks -> ...computation of the Out sets of statements.